src/shims/unix/fd.rs

```diff
@@ -655,47 +664,30 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         Ok(())
     }
 
-    /// Helper to implement `FileDescription::read`:
-    /// `result` should be the return value of some underlying `read` call that used `bytes` as its output buffer.
+    /// Helper to implement `FileDescription::read/write`:
     /// The length of `bytes` must not exceed either the host's or the target's `isize`.
-    /// If `Result` indicates success, `bytes` is written to `buf` and the size is written to `dest`.
-    /// Otherwise, `-1` is written to `dest` and the last libc error is set appropriately.
-    fn return_read_bytes_and_count(
+    /// `is_read` is used to indicate if this helper is used for `read`.
+    /// If it is `read`, `bytes` is written to `buf`.
```

Comment:
No please don't merge the read and write helpers, not for the success case. That's too confusing.

Reply:
Yup, after taking a second look, I realise it is indeed confusing.